app/models/europeana/record/set.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module Europeana
+  class Record
+    # A set of Europeana records
+    #
+    # TODO: store on the Sets API once implemented?
+    class Set < ActiveRecord::Base
+      include HasSettingsAttribute
+
+      self.table_name = 'europeana_record_sets'
+
+      has_one :page_element, dependent: :destroy, as: :positionable
+      has_one :page, through: :page_element
+
+      after_save do
+        page_element&.touch
+      end
+
+      translates :title, fallbacks_for_empty_translations: true
+      accepts_nested_attributes_for :translations, allow_destroy: true
+      default_scope { includes(:translations) }
+
+      validates :europeana_ids, presence: true
+      validates :title, presence: true
+      validate :validate_portal_urls_presence, :validate_portal_urls_format
+
+      has_settings :query_term
+
+      def portal_urls
+        europeana_ids&.map { |id| Europeana::Record.portal_url(id) }
+      end
+
+      # Set Europeana IDs from portal URLs
+      #
+      # @param value [Array<String>]
+      def portal_urls=(value)
+        @portal_urls_to_europeana_ids = value.each_with_object({}) do |url, memo|
+          memo[url] = Europeana::Record.id_from_portal_url(url)
+        end
+        self.europeana_ids = @portal_urls_to_europeana_ids.values.compact
+      end
+
+      def portal_urls_text
+        @portal_urls_text || portal_urls&.join("\n\n")
```

Comment:
when adding a new set this adds a single white space to the text area. It's actually kind of annoying when pasting in urls since they don't validate if theres a leading space. It wouldn't be a problem either if before validating URLs the value has leading and trailing spaces removed down in portal_urls_text=(vlaue)

Reply:
I do not understand what you mean here and/or can not replicate this problematic behaviour.

Reply:
I was referring to the behaviour in rails-admin. I don't think it has anything to do with this line actually after trying to change it here.

To reproduce:
1) Go to the edit page of a browse page.
2) Click the add section button.
3) Observe there's a single empty space in the 'Portal URLs' text area.
4) Add a portal URL to the text area(without removing the space)
5) Click Save
6) Observe that it fails, since the first URL is invalid.

Perhaps this behaviour can be circumvented in the RailsAdmin initializer. I'm not sure how though since it only affects blank values. So that's why trimming the input may be the better solution.